ethereum/eth/src/main/java/org/hyperledger/besu/ethereum/eth/sync/fastsync/FastSyncTargetManager.java

```diff
@@ -67,10 +74,21 @@ public FastSyncTargetManager(
   protected CompletableFuture<Optional<EthPeer>> selectBestAvailableSyncTarget() {
     final BlockHeader pivotBlockHeader = fastSyncState.getPivotBlockHeader().get();
     final Optional<EthPeer> maybeBestPeer = ethContext.getEthPeers().bestPeerWithHeightEstimate();
-    if (!maybeBestPeer.isPresent()) {
-      LOG.debug(
-          "No sync target, checking current peers for usefulness: {}",
-          ethContext.getEthPeers().peerCount());
+    if (maybeBestPeer.isEmpty()) {
+      if (shouldLogDebug.compareAndSet(true, false)) {
```

Comment:
I know log throttling is done in at least one other place: https://github.com/hyperledger/besu/blob/7d5988d4b5aa9db12827f2256fb575ddd3ea6246/ethereum/api/src/main/java/org/hyperledger/besu/ethereum/api/jsonrpc/internal/methods/engine/EngineForkchoiceUpdated.java#L287-L296

Would be good to have a single way to do it in a logging util. I can think of several places we could use it.

Reply:
Has the usage of a [BurstFilter](https://logging.apache.org/log4j/2.x/manual/filters.html#BurstFilter) been analyzed for this use case?

Reply:
I think @fab-10 may have looked into that or some similar feature of log4j. I agree that would be a neater approach and can't see a reason why we wouldn't want to use it for all logs

Reply:
I haven't yet looked at how to throttle this kind of logs, what I did was about summarizing the progress of the sync every tot seconds, this seems different, and I agree that something like the BurstFilter make sense here, since it could also be easily configurable.

Reply:
From my investigation, BurstFilter doesn't appear to be specific enough to address this use case. My understanding is that it throttles the entire log level, whereas this functionality allows a specific (potentially spammy) log message to be throttled. Is my understanding of BurstFilter correct, or is it able to be targeted to a class/method level?

A logging improvement via log4j configuration would likely also require the user to configure it.